code/modules/modular_computers/computers/modular_computer/core.dm

```diff
@@ -23,48 +21,53 @@
 
 // Used to perform preset-specific hardware changes.
 /obj/item/modular_computer/proc/install_default_hardware()
-	return 1
+	var/datum/extension/assembly/assembly = get_extension(src, /datum/extension/assembly)
+	for(var/T in default_hardware)
+		assembly.try_install_component(null, new T(src))
 
 // Used to install preset-specific programs
 /obj/item/modular_computer/proc/install_default_programs()
-	return 1
-
-/obj/item/modular_computer/proc/install_default_programs_by_job(var/mob/living/carbon/human/H)
+	var/mob/living/carbon/human/H = get_holder_of_type(src, /mob)
```

Comment:
okay this is merging two things that shouldn't be really merged this way IMO
default programs by job was called only on custom loadout devices
This makes basic default programs require a human holder (early returns if no human or no job) which means that pre-mapped presets (the main customer of this proc) are allll fucked.
Job programs should be installed on TOP of default ones, not blocking default stuff from existing